src/sessions/softphone-session-handler.ts

```diff
@@ -1,19 +1,93 @@
 import { pick } from 'lodash';
 import { JingleReason } from 'stanza/protocol';
+import { Constants } from 'stanza';
 
 import BaseSessionHandler from './base-session-handler';
-import { IPendingSession, IAcceptSessionRequest, ISessionMuteRequest, IConversationParticipant, IExtendedMediaSession, IUpdateOutgoingMedia, IStartSoftphoneSessionParams } from '../types/interfaces';
-import { SessionTypes, SdkErrorTypes } from '../types/enums';
+import {
+  IPendingSession,
+  IAcceptSessionRequest,
+  ISessionMuteRequest,
+  IConversationParticipant,
+  IExtendedMediaSession,
+  IUpdateOutgoingMedia,
+  IStartSoftphoneSessionParams,
+  IConversationParticipantFromEvent,
+  ICallStateFromParticipant,
+  IStoredConversationState,
+  ISdkConversationUpdateEvent,
+  IConversationHeldRequest
+} from '../types/interfaces';
+import { SessionTypes, SdkErrorTypes, JingleReasons, CommunicationStates } from '../types/enums';
 import { attachAudioMedia, logDeviceChange, createUniqueAudioMediaElement } from '../media/media-utils';
 import { requestApi, isSoftphoneJid, createAndEmitSdkError } from '../utils';
+import { ConversationUpdate } from '../conversations/conversation-update';
+
+type SdkConversationEvents = 'added' | 'removed' | 'updated';
 
 export default class SoftphoneSessionHandler extends BaseSessionHandler {
   sessionType = SessionTypes.softphone;
+  /* Could be active persistent connection or non concurrent session */
+  activeSession?: IExtendedMediaSession;
+  conversations: { [convesationId: string]: IStoredConversationState } = {};
+  lastEmittedSdkConversationEvent: ISdkConversationUpdateEvent = {
+    activeConversationId: undefined,
+    current: [],
+    added: [],
+    removed: []
+  };
+
+  constructor (sdk, handler) {
+    super(sdk, handler);
+    (window as any).softphoneHandler = this; // TODO: take this out
+  }
 
   shouldHandleSessionByJid (jid: string): boolean {
     return isSoftphoneJid(jid);
   }
 
+  handleConversationUpdate (update: ConversationUpdate, sessions: IExtendedMediaSession[]) {
```

Comment:
It would be nice if we can determine whether or not this conversation has softphone stuff on it, and if it doesn't, we should just not handle the update. We need to be careful we don't emit softphone events for a new video conversation or something.

Reply:
That's kind of what I was going for here. Initially, the session-manager handled all the logic, which was pretty simple: 
* If we have a session with this conversationId, pass the update to that session

But this caused me problems because we don't always have a session with the conversationId (if we are getting a new session or the session has ended and we are getting that convo event). So I changed the logic to this: 
* session-manger hands the update to each enabled session-handler with all sessions of that session type. 
* the handlers decided if they should handle it

So for softphone sessions, we will only handle it if the update has callInfo on it (see lines 52 & 57 below). 

Screen-share is no-op and video kept the old logic of looking for a video session with the same conversationId as the update.